packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.ts

```diff
@@ -392,6 +418,22 @@ export default class UserStorageController extends BaseController<
     this.#registerMessageHandlers();
     this.#nativeScryptCrypto = nativeScryptCrypto;
     this.#accounts.setupAccountSyncingSubscriptions();
+
+    // Network Syncing
+    if (env?.isNetworkSyncingEnabled) {
+      startNetworkSyncing({
+        messenger,
+        getStorageConfig: async () => {
+          const { storageKey, bearerToken } =
+            await this.#getStorageKeyAndBearerToken();
+          return {
+            storageKey,
+            bearerToken,
+            nativeScryptCrypto: this.#nativeScryptCrypto,
+          };
+        },
+      });
```

Comment:
I wanted to try to keep this Controller minimal (it is starting to grow in size), so I've made this `startNetworkSyncing` which takes the messenger if it needs to access things inside the controller (IOC).

WDYT?

Reply:
I think it's great, and I'll soon follow the same pattern for account syncing! Nice!

Reply:
PS: But what if you need to trigger network sync from clients? Won't you need a public method for this to be exposed via the controller?

Reply:
Yep debating this too.

## **Event based approach**
We can emit an internal event "BigSyncInProgress", which we can then have all our syncing features subscribe to.

Pros: keeps syncing features decoupled and the UserStorageController minimal
Cons: worried this might be a lot of API calls/computations done in parallel. 

## **Expose a single method that calls the syncing features synchronously**
We have 1 exposed method in the UserStorageController, which then calls internal syncing features

E.g.
```ts
import { performAccountSync } from './account-sync'
import { performNetworkSync } from './network-sync'

public async performSync() {
  await performAccountSync(...)
  await performNetworkSync(...)
}
```

Pro: We can keep a minimal interface
Con: worried that this method would be a long running method to complete. OK on extension (since running in the background), but Mobile might suffer (since there have been some problems with threading before).

We can do some more brainstorming.